dnf-docker-test/features/steps/test_behave.py

```diff
@@ -1,106 +1,69 @@
 #!/usr/bin/python -tt
 
 from behave import *
-import os
-import subprocess
+import dnf
 import glob
-import re
+import os
 import shutil
-from time import sleep
-
-DNF_FLAGS = ['-y', '--disablerepo=*', '--nogpgcheck']
-RPM_INSTALL_FLAGS = ['-Uvh']
-RPM_ERASE_FLAGS = ['-e']
-
-
-def _left_decorator(item):
-    """ Removed packages """
-    return u'-' + item
-
-
-def _right_decorator(item):
-    """ Installed packages """
-    return u'+' + item
-
-
-def find_pkg(pkg):
-    """ Find the package file in the repository """
-    candidates = glob.glob('/repo/' + pkg + '*.rpm')
-    if len(candidates) == 0:
-        print("No candidates for: '{0}'".format(pkg))
-    assert len(candidates) == 1
-    return candidates[0]
-
-
-def decorate_rpm_packages(pkgs):
-    """ Converts package names like TestA, TestB into absolute paths """
-    return [find_pkg(p) for p in pkgs]
-
-
-def get_rpm_package_list():
-    """ Gets all installed packages in the system """
-    pkgstr = subprocess.check_output(['rpm', '-qa', '--queryformat', '%{NAME}\n'])
-    return pkgstr.splitlines()
+import subprocess
+import time
 
 
-def get_rpm_package_version_list():
+def sack_rpm_comparator():
     """ Gets all installed packages in the system with version"""
-    pkgverstr = subprocess.check_output(['rpm', '-qa', '--queryformat', '%{NAME}-%{VERSION}-%{RELEASE}\n'])
-    dnfverstr = subprocess.check_output(['dnf', 'repoquery', '--installed', '-Cq', '--queryformat',
-                                         '%{name}-%{version}-%{release}'])
-    pkgverstr = pkgverstr.splitlines()
-    comparerpmver = pkgverstr
-    for pkg in comparerpmver:
+    # Comparison of epoch is imposible, because rpm return (none), but dnf repoquery 0 if it is not defined.
+    comparerpmver = subprocess.check_output(['rpm', '-qa', '--queryformat', '%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}\n'])
+    dnfverstr = subprocess.check_output(['dnf', 'repoquery', '--installed', '--disableexclude=all', '-Cq', '--queryformat',
+                                         '%{name}-%{version}-%{release}.%{arch}'])
+    comparerpmver = comparerpmver.splitlines()
+    for pkg in list(comparerpmver):
         if pkg.decode().startswith('gpg-pubkey'):
             comparerpmver.remove(pkg)
     assert sorted(comparerpmver) == sorted(dnfverstr.splitlines())
```

Comment:
Ok, here is a solution:
comparerpmver_set = set(comparerpmver)
assert len(comparerpmver) == len(comparerpmver_set)
dnfverstr_set = set(dnfverstr.splitlines())
assert len(dnfverstr_set) == len(dnfverstr.splitlines())
assert dnfverstr_set == comparerpmver_set